File: problems/226-Invert-Binary-Tree/solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* invertTree(struct TreeNode* root){
    if(root == NULL || (root->left == NULL && root->right == NULL)) {
        return root;
    }
    
    struct TreeNode *left = root->left, *right = root->right;
    
    left = invertTree(left);
    right = invertTree(right);
    
    root->left = right;
    root->right = left;
    
    return root;
}

File: problems/226-Invert-Binary-Tree/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}

	left, right := root.Left, root.Right

	invertTree(left)
	invertTree(right)

	root.Left = right
	root.Right = left

	return root
}

